pipeline_tools/confirm_submission.py

```diff
@@ -2,67 +2,81 @@
+from pipeline_tools.http_requests import HttpRequests
 
-RETRY_SECONDS = 10
-TIMEOUT_SECONDS = 600
 
-
-def status_is_invalid(status):
-    return status != 'Valid'
-
-
-@retry(reraise=True, wait=wait_exponential(multiplier=1, max=RETRY_SECONDS), stop=stop_after_delay(TIMEOUT_SECONDS),
-       retry=(retry_if_result(status_is_invalid) | retry_if_exception_type(requests.HTTPError)))
-def wait_for_valid_status(envelope_url):
+def wait_for_valid_status(envelope_url, http_requests):
     """
     Check the status of the submission. Retry until the status is "Valid", or if there is an error with the request to
     get the submission envelope.
 
-    :param str envelope_url: Submission envelope url
-    :return: str status: Status of the submission ("Valid", "Validating", etc.)
-    """
-    print('Getting status for {}'.format(envelope_url))
-    envelope_js = get_envelope_json(envelope_url)
-    status = envelope_js.get('submissionState')
-    print('submissionState: {}'.format(status))
-    return status
+    Args:
+        envelope_url: Submission envelope url
+        http_requests: HttpRequests object
 
+    Returns:
+        Status of the submission ("Valid", "Validating", etc.)
+
+    Raises:
+        requests.HTTPError if response status indicates an error
+    """
+    def log_before(envelope_url):
+        print('Getting status for {}'.format(envelope_url))
+
+    def status_is_invalid(response):
+        envelope_js = response.json()
+        status = envelope_js.get('submissionState')
+        print('submissionState: {}'.format(status))
+        return status != 'Valid'
+
+    response = (
+        http_requests
+        .get(
+            envelope_url,
+            before=log_before(envelope_url),
+            retry=retry_if_result(status_is_invalid)
+        )
+    )
+    return True
+
+
+def confirm(envelope_url, http_requests):
+    """Confirms the submission.
+
+    Args:
+        envelope_url: a string representing the url for the envelope
+        http_requests: HttpRequests object
+
+    Returns:
+        The text of the response
+
+    Raises:
+        requests.HTTPError if the response status indicates an error
+    """
 
-@retry(reraise=True, wait=wait_exponential(multiplier=1, max=RETRY_SECONDS), stop=stop_after_delay(TIMEOUT_SECONDS))
-def confirm(envelope_url):
     print('Confirming submission')
     headers = {
         'Content-type': 'application/json'
     }
-    response = requests.put('{}/submissionEvent'.format(envelope_url), headers=headers)
-    check_status(response.status_code, response.text)
-    print(response.text)
-
-
-def get_envelope_json(envelope_url):
-    response = requests.get(envelope_url)
-    envelope_js = response.json()
-    check_status(response.status_code, response.text)
-    return envelope_js
+    response = http_requests.put('{}/submissionEvent'.format(envelope_url), headers=headers)
+    text = response.text
+    print(text)
```

Comment:
Do we still need to print out the response text to stdout now that we are recording the responses in a separate file?

Reply:
I think it's still worth doing since the recording is optional and might not always be turned on.